File: validatedtextinputlayout/src/main/java/com/julianraj/validatedtextinputlayout/validator/RequiredValidator.java
package com.julianraj.validatedtextinputlayout.validator;

import android.support.annotation.NonNull;

import com.julianraj.validatedtextinputlayout.ValidatedTextInputLayout;

/**
 * Validator to set input field as required.
 * @see BaseValidator
 */
public class RequiredValidator extends BaseValidator {

    /**
     * @param pErrorMessage error message to display if validation fails
     */
    public RequiredValidator(String pErrorMessage) {
        super(pErrorMessage);
    }

    /**
     * @param pErrorMessage error message to display if validation fails
     * @param pCallback callback for validation event
     */
    public RequiredValidator(@NonNull String pErrorMessage, ValidationCallback pCallback) {
        super(pErrorMessage, pCallback);
    }

    /**
     * Check if the associated {@link ValidatedTextInputLayout} is empty or not.
     *
     * @param pText value associated with the input field
     * @return validity of the field
     */
    @Override
    public boolean isValid(String pText) {
        return !pText.isEmpty();
    }
}

File: validatedtextinputlayout/src/main/java/com/julianraj/validatedtextinputlayout/validator/RequiredValidator.kt
package com.julianraj.validatedtextinputlayout.validator

import com.julianraj.validatedtextinputlayout.ValidatedTextInputLayout

/**
 * Validator to set input field as required.
 *
 * @param errorMessage Error message that will be displayed if validation fails.
 * @param callback callback with validation result
 *
 * @see BaseValidator
 */
class RequiredValidator(errorMessage: String,
                        callback: ValidationCallback? = null
) : BaseValidator(errorMessage, callback) {

    /**
     * Check if the associated [ValidatedTextInputLayout] is empty or not.
     *
     * @param text value associated with the input field
     * @return validity of the field
     */
    override fun isValid(text: String): Boolean {
        return text.isNotEmpty()
    }
}
